eng/source-build-patches/0002-Use-package-version-properties-for-reference-version.patch

```diff
@@ -1,29 +0,0 @@
-From 74c9b072853ff4d68f2630462a6f1ac5d576979d Mon Sep 17 00:00:00 2001
-From: Chris Rummel <crummel@microsoft.com>
-Date: Mon, 21 Sep 2020 09:59:02 -0500
-Subject: [PATCH 2/2] Use package version properties for reference versions.
-
----
- eng/Versions.props                                         | 7 +++++--
- src/System.CommandLine/System.CommandLine.csproj           | 4 ++--
- 2 files changed, 7 insertions(+), 4 deletions(-)
-
-diff --git a/eng/Versions.props b/eng/Versions.props
-index 3048701..16a0eb7 100644
---- a/eng/Versions.props
-+++ b/eng/Versions.props
-@@ -29,9 +29,12 @@
-     <UsingToolSymbolUploader>true</UsingToolSymbolUploader>
-     <PublishWindowsPdb>true</PublishWindowsPdb>
-   </PropertyGroup>
--  <!--   <PropertyGroup>
-+  <PropertyGroup>
-     <FluentAssertionsVersion>4.18.0</FluentAssertionsVersion>
-     <NewtonsoftJsonVersion>9.0.1</NewtonsoftJsonVersion>
-     <SystemNetHttpVersion>4.3.1</SystemNetHttpVersion>
--  </PropertyGroup>  -->
-+    <MicrosoftExtensionsHostingPackageVersion Condition="'$(MicrosoftExtensionsHostingPackageVersion)' == ''">2.2.0</MicrosoftExtensionsHostingPackageVersion>
```

Comment:
Have you verified the removal of this patch and others aren't going to introduce prebuilts?  They were added for a reason.

Reply:
My mistake. My impression was that the .slnf approach was a replacement for the patch files.

To make this easier going forward, if patches are needed, I would prefer we add:

* a way to reproduce them easily so that contributors unfamiliar with the process don't get blocked or update them incorrectly, and
* an automated test that verifies that they're producing exactly what's expected.

I'm also not clear on the reasons why some packages are included and others aren't. 

@omajid, any insights here?

Reply:
I went ahead and pulled this commit and performed a full source-build to verify no prebuilts were introduced.  Everything built cleanly although there was an api change affecting runtime.

The goal is to eliminate all source-build patches.  Any changes necessary for source-build should be part of the product.  We are also working to catch as many issues as possible in the source-build leg of the repo builds.  The last big remaining work is to enable prebuilt detection.  This was something we wanted to get in for 6.0 but couldn't fit it in so it will happen in the 7.0 timeframe.  Once this is complete the dev UX for source-build should be much better.  Thanks for your patience and thank you Jon for integrating the patches here and removed the unneeded ones.